Sources/PointFree/Account/Change.swift

```diff
@@ -12,82 +12,72 @@ import PointFreeRouter
 import Prelude
 import Stripe
 import Styleguide
+import TaggedMoney
 import Tuple
 import View
 
 // MARK: Middleware
 
-let subscriptionChangeMiddleware =
+let subscriptionChangeMiddleware: Middleware<
+  StatusLineOpen,
+  ResponseEnded,
+  Tuple2<User?, Pricing?>,
+  Data
+  > =
   filterMap(require1 >>> pure, or: loginAndRedirect)
-    <<< filterMap(
-      require2 >>> pure,
-      or: redirect(
-        to: .account(.index),
-        headersMiddleware: flash(
-          .error,
-          "Invalid subscription data. Please try again or contact <support@pointfree.co>."
-        )
-      )
-    )
+    <<< filterMap(require2 >>> pure, or: invalidSubscriptionErrorMiddleware)
     <<< fetchSeatsTaken
     <<< requireStripeSubscription
     <<< requireActiveSubscription
     <<< requireValidSeating
-    <| map(lower)
-    >>> subscriptionChange
+    <| changeSubscription(
```

Comment:
I generalized `subscriptionChange` so that it can take custom middleware to run in the success and failure cases of modifying the subscription. this allowed me to reuse all of this logic easily for my new route